lib/namespace.js

```diff
@@ -187,9 +187,8 @@ Namespace.prototype.add = function(client, fn){
  */
 
 Namespace.prototype.remove = function(socket){
-  var i = this.sockets.indexOf(socket);
-  if (~i) {
-    this.sockets.splice(i, 1);
+  if (this.sockets[socket.id]) {
```

Comment:
`this.sockets.hasOwnProperty` would be a better test. 
See for inspiration: https://github.com/socketio/socket.io-adapter/pull/30